### PR TITLE
feat: add json data provider

### LINKED ISSUE #1: Create first iteration with simple UI
### What should we do?

Allow data to be provided through examples. It should

- add a section in the properties panel
- allow pasting JSON into it
- extract the schema and types from the examples
- Suggest them in the FEEL editor

### Why should we do it?

This enables us to deliver the data schema support as a MVP with the 8.2 release.

related to https://github.com/bpmn-io/internal-docs/issues/680

### LINKED ISSUE #2: React to Element Template bindings
### What should we do?

When an Element Template is applied, still show the data section, unless the element template binds to the property itself

### Why should we do it?

So example data seamlessly integrates into the connectors story

### PR DESCRIPTION
**Pull request by marstamm**
This Module includes a variable provider and a properties panel extension that allows you to add external variables through JSON.

Recording 2023-03-24 at 14 23 04

Test it out with `npm start`.

Follow-ups:
 - Input validation in the properties panel: https://github.com/bpmn-io/bpmn-js-json-data-provider/issues/3
 - Element template integration: https://github.com/bpmn-io/bpmn-js-json-data-provider/issues/2

closes #1
closes #2
related to https://github.com/bpmn-io/internal-docs/issues/680

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,54 @@
 {
-  "name": "bpmn-js-example-data-extension",
+  "name": "@bpmn-io/bpmn-js-json-data-provider",
```
Comment (nikku) [inline package.json:2]: :+1: 

Good to namespace an experimental (micro) utility.

Comment (nikku) [reply package.json:2]: However I'm wondering if we could make more clear what it does:

* It is specific to Camunda, and adds example data facilities there.

**Review by nikku — COMMENTED** [review]
Looks reasonable to me. Two questions:

* what shall this do, and hence what is the appropriate name? `@camunda/example-data-properties-provider` could be a clear, alternative name
* We don't seem to hook into element templates yet (if example data is provided as a _hidden_ property

**Comment by marstamm** [comment]
> what shall this do, and hence what is the appropriate name? @camunda/example-data-properties-provider could be a clear, alternative name

That's reasonable, let's adjust it. I'll also look into moving it to the `camunda` github organization as well

> We don't seem to hook into element templates yet (if example data is provided as a hidden property)?

Correct, currently we always show the group. I wanted to get the MVP done for the 8.2 release, so get the core reviewed early. #2 is a relatively small follow-up which we can tackle next week and release it if we have enough time

**Comment by marstamm** [comment]
I changed the namespace with [179c27c](https://github.com/bpmn-io/bpmn-js-json-data-provider/pull/4/commits/179c27ccd222c5f5c9287487372b1905bbab495b). I could not transfer the github repo, as I don't have permissions to create public repos in the camunda org. I will reach out to get it transferred.

This should not block us from merging, as npm orgs are independent of github orgs.

**Comment by marstamm** [comment]
Also added the element template integration in a follow-up: https://github.com/bpmn-io/bpmn-js-json-data-provider/pull/5

**Comment by marstamm** [comment]
@nikku do you have time to have another look at the changes or should I onboard another reviewer?

**Pull request merged**